eth/api.go

```diff
@@ -362,9 +362,10 @@ func generateWitness(blockchain *core.BlockChain, block *types.Block) (*stateles
 	// Collect storage locations that prover needs but sequencer might not touch necessarily
 	statedb.GetState(rcfg.L2MessageQueueAddress, rcfg.WithdrawTrieRootSlot)
 
-	// Note: scroll-revm detects the Feynman transition block using this storage slot,
-	// since it does not have access to the parent block timestamp. We need to make
-	// sure that this is always present in the execution witness.
+	// Note: scroll-reth detects the Feynman transition block using the block
+	// timestamp, but since there might be multiple blocks with the same timestamp,
+	// it also needs this value to avoid applying the state update multiple times.
+	// So we need to make sure that this is always present in the execution witness.
 	statedb.GetState(rcfg.L1GasPriceOracleAddress, rcfg.IsFeynmanSlot)
```

Comment:
You still include the witness for every block?

Reply:
> it also needs this value to avoid applying the state update multiple times.

why set this can avoid the multiples applying

Reply:
> why set this can avoid the multiple applying, and what's the result after applying many times

I think it's because there's only one time that a block is in `feynman` time and IsFeynmanSlot is `false`, the later blocks with `feynman` time will have IsFeynmanSlot `true`.